src/core/atomic.d

```diff
@@ -1273,6 +1182,146 @@ else version( AsmX86_64 )
     }
 }
 
+/* ***
+ * Functions built in to the compiler, along with a default implementation.
+ */
+
+bool cmpxchg(shared(ubyte)* here, ubyte ifThis, ubyte writeThis) pure nothrow @nogc @trusted
```

Comment:
If this doesn't offer anything extra over cas then there's no point in having it public.